modules/layers/src/column-layer/column-layer.js

```diff
@@ -89,83 +101,147 @@ export default class ColumnLayer extends Layer {
+      props.wireframe !== oldProps.wireframe;
+
+    if (regenerateModels) {
       const {gl} = this.context;
-      if (this.state.model) {
-        this.state.model.delete();
+      if (this.state.models) {
+        this.state.models.forEach(model => model.delete());
       }
-      this.setState({model: this._getModel(gl)});
+      this.setState(this._getModels(gl));
       this.getAttributeManager().invalidateAll();
     }
-
     if (props.vertices !== oldProps.vertices) {
       this._updateVertices(props.vertices);
     }
   }
 
-  getGeometry(diskResolution) {
-    return new CylinderGeometry({
+  getGeometry(diskResolution, mode = FILL_MODE, drawMode = GL.TRIANGLES) {
+    return new ColumnGeometry({
       radius: 1,
       topCap: false,
       bottomCap: true,
+      mode,
+      drawMode,
       height: 2,
       verticalAxis: 'z',
       nradial: diskResolution,
       nvertical: 1
     });
   }
 
-  _getModel(gl) {
-    return new Model(
-      gl,
-      Object.assign({}, this.getShaders(), {
-        id: this.props.id,
-        geometry: this.getGeometry(this.props.diskResolution),
-        isInstanced: true,
-        shaderCache: this.context.shaderCache
-      })
-    );
+  _getModels(gl) {
+    const {id, filled, wireframe, diskResolution} = this.props;
+
+    let filledModel;
+    let strokeModel;
+
+    if (filled) {
+      filledModel = new Model(
+        gl,
+        Object.assign({}, this.getShaders(), {
+          id: `${id}-top`,
+          uniforms: {isWireframe: false},
+          geometry: this.getGeometry(diskResolution, FILL_MODE),
+          isInstanced: true,
+          shaderCache: this.context.shaderCache
+        })
+      );
+    }
+    if (wireframe) {
+      strokeModel = new Model(
+        gl,
+        Object.assign({}, this.getShaders(), {
+          id: `${id}-side`,
+          uniforms: {isWireframe: true},
+          geometry: this.getGeometry(diskResolution, OUTLINE_MODE, GL.LINES),
+          isInstanced: true,
+          shaderCache: this.context.shaderCache
+        })
+      );
+    }
+
+    return {
+      models: [strokeModel, filledModel].filter(Boolean),
+      filledModel,
+      strokeModel
+    };
   }
 
   _updateVertices(vertices) {
     if (!vertices) {
       return;
     }
-
     const {diskResolution} = this.props;
     log.assert(vertices.length >= diskResolution);
+    const {filledModel, strokeModel} = this.state;
 
-    const {model} = this.state;
-    const geometry = this.getGeometry(this.props.diskResolution);
-    const positions = geometry.attributes.POSITION;
-    let i = 0;
-    for (let loopIndex = 0; loopIndex < 3; loopIndex++) {
-      for (let j = 0; j <= diskResolution; j++) {
-        const p = vertices[j] || vertices[0]; // auto close loop
-        // replace x and y in geometry
-        positions.value[i++] = p[0];
-        positions.value[i++] = p[1];
-        i++;
+    const strokeGeometry = this.getGeometry(diskResolution, OUTLINE_MODE, GL.LINES);
```

Comment:
update `fillGeometry` positions should happen when `fillModel` exists.

Let's be consistent, use `fill` instead of `filled`, use `stroke` instead of `outline`, `OUTLINE_MODE` -> `STROKE_MODE`